src/shared/istiopolicy/creator.go

```diff
@@ -137,11 +137,10 @@ func (c *Creator) updateOrCreatePolicy(
 }
 
 func (c *Creator) isPolicyEqual(existingPolicy *v1beta1.AuthorizationPolicy, newPolicy *v1beta1.AuthorizationPolicy) (bool, error) {
-	if existingPolicy.Spec.Selector == nil || newPolicy.Spec.Selector == nil {
-		return false, fmt.Errorf("policy pod selector is nil")
-	}
-
-	if len(existingPolicy.Spec.Rules) == 0 || len(existingPolicy.Spec.Rules[0].From) == 0 {
+	if len(existingPolicy.Spec.Rules) == 0 ||
+		len(existingPolicy.Spec.Rules[0].From) == 0 ||
+		existingPolicy.Spec.Selector == nil ||
+		len(existingPolicy.Spec.Selector.MatchLabels) == 0 {
 		logrus.Warning("found existing policy with bad format, overwriting")
```

Comment:
When does this happen? What does bad format mean? If it is possible to create a bad policy, we should change that flow so it's no longer possible. I think all flows should either end in success or error, logging a warning probably means something we would rather not happen, happened.

Reply:
If it happens it's not because of us, but a result of some manually created policy. 
If there are no Rules we would crush due to nil reference, and like any output we receive from outer-space, it can't kill us.

Reply:
Ok, I'll change that